app/models/cart.rb

```diff
@@ -35,6 +51,13 @@ def discount
   end
 
   def order_total
-    coupon ? sub_total - discount : sub_total
+    price_with_coupon = coupon ? sub_total - discount : sub_total
+    delivery ? price_with_coupon + delivery.price : sub_total
+  end
+
+  private
+
+  def set_number_and_status
+    update(number: "R#{id.to_s.rjust(8, '0')}", status: 0)
```

Comment:
since `status` is integer field, you can define `default: 0` right in migrations.

Reply:
removed status update from class method to migration as a default value